authentication/src/main/java/uk/gov/android/authentication/LoginSession.kt

```diff
@@ -0,0 +1,100 @@
+package uk.gov.android.authentication
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.ActivityCompat
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+import net.openid.appauth.TokenResponse
+import java.util.UUID
+
+class LoginSession : ILoginSession {
+    private var context: Context? = null
+    private lateinit var authService: AuthorizationService
+
+    override fun init(
+        context: Context
+    ): ILoginSession {
+        if (this.context == null) {
+            this.context = context
+            authService = AuthorizationService(context)
+        }
+        return this
+    }
```

Comment:
This sort of goes into one of the preceding comments - if it's worth using an Injectable context over the use of this function and `AuthorizationService` instantiation?

Reply:
See above reply